app/models/authentication.rb

```diff
@@ -81,6 +81,18 @@ def raise_event(status, _message = nil)
     MiqEvent.raise_evm_event_queue(ci, "#{prefix}_auth_#{status}")
   end
 
+  def assign_values(options)
+    options.each do |key, value|
+      next unless Authentication.column_names.include?(key) && value
+      if self[key].kind_of? Array
+        self[key] << value
```

Comment:
if value is an Array itself, this would not append an Array to another Array.  Are there cases where you would even pass value as an Array?